app/components/pdfViewer/index.tsx

```diff
@@ -1,146 +1,140 @@
-        </div>
-      );
-    }
-    return null;
-  }
+const PDFViewer: React.FunctionComponent<PDFViewerProps> = props => {
+  const { pdfURL, shouldShow, onFailed, onLoadSuccess, filename } = props;
+  const [isFetching, setIsFetching] = React.useState(false);
+  const [PDFBinary, setPDFBinary] = React.useState(null);
+  const [extend, setExtend] = React.useState(false);
+  const [hadErrorToLoad, setLoadError] = React.useState(false);
+  const [succeedToLoad, setSucceed] = React.useState(false);
+  const [pageCountToShow, setPageCountToShow] = React.useState(0);
+  const wrapperNode = React.useRef<HTMLDivElement | null>(null);
 
-  private getContent = () => {
-    const { isFullNode, numPages } = this.state;
+  React.useEffect(
+    () => {
+      if (shouldShow) {
+        setIsFetching(true);
+        Axios.get(
+          `https://u5ctiiqdab.execute-api.us-east-1.amazonaws.com/dev/get-pdf?pdf_url=${pdfURL}&title=${filename}`,
+          {
+            responseType: "blob",
+          }
+        )
+          .then(res => {
+            setPDFBinary(res.data);
+            setIsFetching(false);
+          })
+          .catch(_err => {
+            setIsFetching(false);
+          });
+      }
+    },
+    [pdfURL]
+  );
 
-    if (isFullNode) {
-      return Array.from(new Array(numPages), (_el, index) => <Page key={`page_${index + 1}`} pageNumber={index + 1} />);
+  const getContent = () => {
+    if (extend) {
+      return Array.from(new Array(pageCountToShow), (_el, index) => (
+        <Page key={`page_${index + 1}`} pageNumber={index + 1} />
+      ));
     } else {
       return <Page pageNumber={1} />;
     }
   };
 
-  private onDocumentLoadSuccess = (document: any) => {
-    const { numPages } = document;
-    this.setState({
-      numPages,
-    });
-  };
-}
+  if (isFetching) {
+    return (
+      <div className={styles.loadingContainer}>
+        <ArticleSpinner />
+      </div>
+    );
+  }
 
-function mapStateToProps(appState: AppState) {
-  return {
-    layout: appState.layout,
-  };
-}
+  if (shouldShow && PDFBinary) {
+    return (
+      <div ref={wrapperNode}>
+        <Document
+          // tslint:disable-next-line:max-line-length
+          file={PDFBinary}
+          error={null}
+          loading={
+            <div className={styles.loadingContainer}>
+              <ArticleSpinner />
+            </div>
+          }
+          onLoadError={() => {
+            setLoadError(true);
+            onFailed();
+          }}
+          onLoadSuccess={(document: any) => {
+            setPageCountToShow(document.numPages);
+            setSucceed(true);
+            onLoadSuccess();
+          }}
+        >
+          {getContent()}
+        </Document>
+
+        <div style={{ display: "flex", justifyContent: "center", marginTop: "16px", marginBottom: "65px" }}>
+          {succeedToLoad && (
+            <>
+              <ScinapseButton
+                style={{
+                  display: "flex",
```

Comment:
This section seems to overlap with the code below.
How about using the code below?
~~~
const style: React.CSSProperties ={
                  display: "flex",
                  justifyContent: "center",
                  alignItems: "center",
                  backgroundColor: "#3e7fff",
                  width: "154px",
                  height: "40px",
};
const style2: React.CSSProperties = {...style, ....any code};